demos/python_demos/audio_classification_demo/audio_classification_demo.py

```diff
@@ -0,0 +1,216 @@
+        data = wav.readframes(params.nframes)
+        if sampwidth_types.get(params.sampwidth):
+            data = np.frombuffer(data, dtype=sampwidth_types[params.sampwidth])
+        else:
+            raise RuntimeError("Couldn't process file {}: unsupported sample width {}"
+                               .format(str(file), params.sampwidth))
+        data = np.reshape(data, (params.nframes, params.nchannels))
+        if as_float:
+            data = data / sampwidth_max[params.sampwidth]
+
+    return params.framerate, data
+
+
+def main():
+    args = build_argparser()
+
+    logging.basicConfig(format="[ %(levelname)s ] %(message)s", level=logging.INFO, stream=sys.stdout)
+    log = logging.getLogger()
+
+    log.info("Creating Inference Engine")
+    ie = IECore()
+
+    if args.device == "CPU" and args.cpu_extension:
+        ie.add_extension(args.cpu_extension, 'CPU')
+
+    log.info("Loading model {}".format(args.model))
+    net = ie.read_network(args.model, args.model[:-4] + ".bin")
+
+    if args.device == "CPU":
+        supported_layers = ie.query_network(net, args.device)
+        not_supported_layers = [l for l in net.layers.keys() if l not in supported_layers]
+        if len(not_supported_layers) > 0:
+            log.error("Following layers are not supported by the {} plugin:\n {}"
+                      .format(args.device, ', '.join(not_supported_layers)))
+            log.error("Please try to specify cpu extensions library path in sample's command line parameters using "
+                      "-l/--cpu_extension command line argument")
+            sys.exit(1)
+
+    if len(net.inputs) != 1:
+        log.error("Demo supports only models with 1 input layer")
+        sys.exit(1)
+    input_blob = next(iter(net.inputs))
+    input_shape = net.inputs[input_blob].shape
+    if len(net.outputs) != 1:
+        log.error("Demo supports only models with 1 output layer")
+        sys.exit(1)
+    output_blob = next(iter(net.outputs))
+
+    log.info("Loading model to the plugin")
+    exec_net = ie.load_network(network=net, device_name=args.device)
+
+    log.info("Preparing input")
+
+    labels = []
+    if args.labels:
+        with open(args.labels, "r") as file:
+            labels = [l.rstrip() for l in file.readlines()]
+
+    batch_size, channels, _, length = input_shape
+
+    audio = AudioSource(args.input, channels=channels, samplerate=args.samplerate)
+    audio.load()
+
+    hop = length - args.overlap if isinstance(args.overlap, int) else int(length * (1.0 - args.overlap))
+    if hop < 0:
+        log.error("Wrong value for '-ol/--overlap' argument - overlapping more than clip length")
+        sys.exit(1)
+
+    log.info("Starting inference")
+    outputs = []
+    clips = 0
+    infer_time = 0
+    for idx, chunk in enumerate(audio.chunks(length, hop, num_chunks=batch_size)):
+        if len(chunk.shape) != len(input_shape):
+            chunk = np.reshape(chunk, newshape=input_shape)
```

Comment:
I think you could just replace this with `chunk.shape = input_shape` and let NumPy figure it out.